src/contracts/gasCalculator/BaseGasCalculator.sol

```diff
@@ -0,0 +1,63 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.25;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IL2GasCalculator } from "src/contracts/interfaces/IL2GasCalculator.sol";
+
+/// @notice Implementation:
+/// https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L2/GasPriceOracle.sol
+/// @notice Deployment on Base: https://basescan.org/address/0x420000000000000000000000000000000000000f
+interface IGasPriceOracle {
+    function getL1FeeUpperBound(uint256 _unsignedTxSize) external view returns (uint256);
+}
+
+contract BaseGasCalculator is IL2GasCalculator, Ownable {
+    uint256 internal constant _CALLDATA_LENGTH_PREMIUM = 32;
+    uint256 internal constant _BASE_TRANSACTION_GAS_USED = 21_000;
+
+    address public immutable gasPriceOracle;
+    int256 public calldataLengthOffset;
+
+    constructor(address _gasPriceOracle, int256 _calldataLengthOffset) Ownable(msg.sender) {
+        gasPriceOracle = _gasPriceOracle;
+        calldataLengthOffset = _calldataLengthOffset;
+    }
```

Comment:
In the latest Atlas style change we use the `_` prefix to show local variables and no prefix to show function args. Its a bit annoying sometimes with name collisions like "calldataLengthOffset" but maybe we can do something like this?
```suggestion
    constructor(address gasPriceOracle, int256 calldataLenOffset) Ownable(msg.sender) {
        GAS_PRICE_ORACLE = gasPriceOracle;
        calldataLengthOffset = calldataLenOffset;
    }
```